docs/build/build_system.md

```diff
@@ -0,0 +1,71 @@
+# Build system
+
+The scp-machine build system is based on the principles of the [sc-machine](https://github.com/ostis-ai/sc-machine) build system. To learn more about them, read [sc-machine build system documentation](https://ostis-ai.github.io/sc-machine/build/build_system/).
+
+## Compilation steps
+
+This project uses CMake presets. These presets help manage different build configurations, making it easier to work with dependencies and streamline the build process.
+
+If you wish to use CLI instead, you can list all the available configurations in the `CMakePresets.json` file and pick one of the presets:
+
+```sh
+cmake --list-presets
+```
+
+### Configure presets
+
+Configure presets define how the project is configured before building. They specify various settings, including whether to include tests and which dependencies to use.
+
+| **Name**                     | **Display Name**                        | **Description**                                      |
+|------------------------------|-----------------------------------------|------------------------------------------------------|
+| `debug-conan`                | Debug config (Conan)                    | Debug config with tests (Conan dependencies used)    |
+| `release-conan`              | Release config (Conan)                  | Release config (Conan dependencies used)             |
+| `release-with-tests-conan`   | Release config with tests (Conan)       | Release config with tests (Conan dependencies used)  |
+
+## Build presets
+
+Build presets link to the configure presets and specify how to build the project. They determine the output directories based on the selected configuration.
+
+| **Name**   | **Configure Preset**                        |
+|------------|---------------------------------------------|
+| `debug`    | `debug-conan`                               |
+| `release`  | `release-conan`, `release-with-tests-conan` |
+
+The build artifacts will be located in the `build/<Release|Debug>/bin` folder.
+
+## Recommendations
+
+- For development and debugging, use the **Debug config (Conan)** (`debug-conan`) preset. This configuration includes options for testing and benchmarking.
+- For production builds, choose the **Release config (Conan)** (`release-conan`) preset to create optimized builds without debugging information.
+- If you need to run tests in a production build, select the **Release config with tests (Conan)** (`release-with-tests-conan`) preset.
+
+## Example usage
+
+To build your project using these presets, you can use the following commands:
+
+```sh
+# Use pipx to install Conan if not already installed
+pipx install conan
+```
+
+```sh
+# Install dependencies using Conan with Debug build type
+conan install . -s build_type=Debug
```

Comment:
![image](https://github.com/user-attachments/assets/3e269d91-1ef8-4f3e-9c8d-c26fce9ce373)

Reply:
Recheck, please

Reply:
conanfile.py requires sc-machine 0.10.0 and sc-machine has version 0.9.0 in CMakeLists.txt
same error for 0.10.0
when I change to 0.9.0 I get another error
![image](https://github.com/user-attachments/assets/2a290d22-3154-4250-9727-d97577558d10)

Reply:
Try to build packages locally from sources